### PR TITLE
Added manage.py cms import-library functionality

### PR DESCRIPTION
**Pull request by hetmantsev**
Add ability to import course libraries automatically with scripts. (needed for mc courses).

cherry-pick from f94048213e49d1326519930755bb45d018a3faeb.

Tested on few mc projects.

Using: 
`sudo -u www-data /edx/bin/python.edxapp /edx/bin/manage.edxapp cms --settings=aws import-library $LIBRARY_PATH`

### CODE REVIEW

**Comment by hetmantsev** [comment]
Tested on https://edx-staging-mc-import.raccoongang.com/

Import log: https://jenkins.raccoongang.com/job/Import-Microsoft-courses/9/console

File: cms/djangoapps/contentstore/management/commands/tests/test_import_library.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+"""
+Unittests for importing a library via management command
+"""
+
+import ddt
+import os
+from path import Path as path
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/tests/test_import_library.py]: https://github.com/raccoongang/edx-platform/pull/205/files#diff-43462fc4509a219f009dd5f13543cd99R44

File: cms/djangoapps/contentstore/management/commands/tests/test_import_library.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import shutil
+import tempfile
+from StringIO import StringIO
+
+from django.core.management import CommandError, call_command
+
+from opaque_keys.edx.keys import CourseKey
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/tests/test_import_library.py]: fixed.

File: cms/djangoapps/contentstore/management/commands/import-library.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+from optparse import make_option
+
+from django.core.management.base import BaseCommand, CommandError
+
+from django_comment_common.utils import are_permissions_roles_seeded, seed_permissions_roles
+from xmodule.contentstore.django import contentstore
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.xml_importer import import_library_from_xml
+import os
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/import-library.py]: fixed.

File: cms/djangoapps/contentstore/management/commands/import-library.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+class Command(BaseCommand):
+    """
+    Import the specified library data directory into the default ModuleStore
+    """
+    help  = 'Import the specified library(s) data directory into the default ModuleStore\n\n'
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/import-library.py]: fixed.

File: cms/djangoapps/contentstore/management/commands/import-library.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    help += '                     directory and its content must be readable by www-data user\n'
+  
+    option_list = BaseCommand.option_list + (
+        make_option('--nostatic',
+                    action='store_true',
+                    help='Skip import of static content'),
+    )
+  
+    def handle(self, *args, **options):
+        "Execute the command"
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/import-library.py]: fixed.

File: cms/djangoapps/contentstore/management/commands/import-library.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        course_items = import_library_from_xml(
+            mstore, ModuleStoreEnum.UserID.mgmt_command, data_dir, source_dirs, load_error_modules=False,
+            static_content_store=contentstore(), verbose=True,
+            do_import_static=do_import_static,
+            create_if_not_present=True,
+        )
+ 
+        if course_items:
+            self.stdout.write("Successfully imported {} libraries.\n".format(
+                len(course_items)))
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/import-library.py]: fixed.

File: cms/djangoapps/contentstore/management/commands/tests/test_import_library.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        """
+        Try to import library
+        """
+        with modulestore().default_store(ModuleStoreEnum.Type.split):
+            call_command('import-library', self.content_dir, self.library_dir)
+            self.library_from_store = modulestore().get_library(self.library_key)
+
+        # compare library_key of imported library with self-generated
+        self.assertEqual(unicode(self.library_from_store.location.library_key), unicode(self.library_key))
+        # conpare display_name of imported library with original
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/tests/test_import_library.py]: fixed.

File: cms/djangoapps/contentstore/management/commands/import-library.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        if len(args) > 1:
+            source_dirs = args[1:]
+        else:
+            raise CommandError("ERROR: import-library requires at least one library_dir as argument")
+ 
+        self.stdout.write("Importing.  Data_dir={data}, source_dirs={courses}\n".format(
+            data=data_dir,
+            courses=source_dirs,
+        ))
+        mstore = modulestore()
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/import-library.py]: fixed.

File: cms/djangoapps/contentstore/management/commands/tests/test_import_library.py [hunk 9]
```diff
@@ -0,0 +1,69 @@
+"""
+Unittests for importing a library via management command
+"""
+
+import ddt
+import os
+from path import Path as path
+import shutil
+import tempfile
+from StringIO import StringIO
```
[the remark this answers is not in the document]
Comment (hetmantsev) [reply cms/djangoapps/contentstore/management/commands/tests/test_import_library.py]: fixed.

File: cms/djangoapps/contentstore/management/commands/tests/test_import_library.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rimikt) [reply cms/djangoapps/contentstore/management/commands/tests/test_import_library.py]: :)

**Pull request merged**